src/index.ts

```diff
@@ -50,16 +48,13 @@ export function renderAst(
       try {
         buildDom(element);
       } catch (e) {
-        tagErrors.push(e.message);
+        const errorMsg = e.message as string;
+        return {
```

Comment:
Suggestion: let this function take `reportError`. Normally/for now, it can be something like
```javascript
function reportError(tag, err) {
  err.message = `[${tag}] ${err.message}`;
  throw err;
}
```
and it will fail-fast and bubble-up. Separately, another debugging entry-point could pass something like
```javascript
const errors = [];
const result = renderAst(tree, instructions, (tag, err) => errors[tag].concat(err)); // plus some handling new entries
return {...result, errors}; // or whatever this might look like
```

That way only the calling code needs to care about the shape of the result there

Reply:
Going to change this a bit. Instead of just returning `e.message` we should pass up the stack as well.

Reply:
May as well pass the whole error object around

Reply:
> May as well pass the whole error object around

I think I'm doing this now.